pkg/controller/queuejobresources/utils.go

```diff
@@ -31,20 +31,43 @@ limitations under the License.
 package queuejobresources
 
 import (
+	"strings"
+
 	clusterstateapi "github.com/IBM/multi-cluster-app-dispatcher/pkg/controller/clusterstate/api"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/klog/v2"
 )
 
 // filterPods returns pods based on their phase.
 func FilterPods(pods []*v1.Pod, phase v1.PodPhase) int {
-        result := 0
-        for i := range pods {
-                if phase == pods[i].Status.Phase {
-                        result++
-                }
-        }
-        return result
+	result := 0
+	for i := range pods {
+		if phase == pods[i].Status.Phase {
+			result++
+		}
+	}
+	return result
+}
+
+//check if pods pending are failed scheduling
+func PendingPodsFailedSchd(pods []*v1.Pod) map[string][]v1.PodCondition {
+	var podCondition = map[string][]v1.PodCondition{}
```

Comment:
The caller function checks the size of this return value thus to avoid a  nil pointer exception good to create an empty map of lenthg 0, e.g. `make(map[int]v1.PodCondition{})`